ci/smoke-test.bash

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -ex
+
+export RUST_BACKTRACE=1
+
+rustc -vV
+cargo -vV
+
+FEATURES=('--no-default-features' '--features' 'curl-backend,reqwest-backend,reqwest-default-tls')
+case "$(uname -s)" in
+  *NT* ) ;; # Windows NT
+  * ) FEATURES+=('--features' 'vendored-openssl') ;;
+esac
+
+case "$TARGET" in
+  # these platforms aren't supported by ring:
+  powerpc* ) ;;
+  mips* ) ;;
+  riscv* ) ;;
+  s390x* ) ;;
+  aarch64-pc-windows-msvc ) ;;
+  # default case, build with rustls enabled
+  * ) FEATURES+=('--features' 'reqwest-rustls-tls') ;;
+esac
+
+# rustc only supports armv7: https://doc.rust-lang.org/nightly/rustc/platform-support.html
+if [ "$TARGET" = arm-linux-androideabi ]; then
+  export CFLAGS='-march=armv7'
+fi
+
+cargo build --locked --release --target "$TARGET" "${FEATURES[@]}"
+
+mkdir home
+export RUSTUP_HOME=home 
+export CARGO_HOME=home
+
+target/"$TARGET"/release/rustup-init --no-modify-path  --default-toolchain=none --profile default -y
+# shellcheck source=home/env
+source home/env
+rustup toolchain install --profile=default nightly-2022-02-23
+
+rustc --version
+rustdoc --version
+cargo --version
+rustfmt --version
+cargo-fmt --version
+cargo-clippy --version
+clippy-driver --version
+# rust-lldb --version
+# rust-gdb -h
+# rust-gdbgui -h
```

Comment:
Why not running these ?

Reply:
Because they have other dependencies and there are related issues that have not been resolved. So we can skip it for now.
cc: https://github.com/rust-lang/rustup/issues/2838 and https://github.com/rust-lang/rust/issues/61282